### DISCUSSION TITLE
Dynamic certificate pinning

### DISCUSSION THREAD

**Discussion by gabsamples6** [ZitchCode/SecureHttpClient#22] [General]
Hi ,
just wondering if this nuget handles dynamic certificate pinning, I mean the ability to change a pin without the need to release a new version of the app.

Many thanks

**Comment by tranb3r** [comment]
Hi,

Here is how I'm doing it:
- store the certificate pin in the app configuration
- store the app configuration in the cloud (for example in firebase remote config)

The app can easily fetch a new configuration (with the new certificate pin value) without having to release a new version of the app.

Hope this helps.

**Reply by gabsamples6** [reply]
thanks for your reply. sounds good. I am looking for something that does just that and this seems to fit the bill.. Do you know if this package is been used in Maui - Productions by any chance..  Also I guess to use Firebase remote config we would have to use some other nugets that probably uses  Xamarin.firebase components that they have not updated in years on ios side...
Many thanks for replying anyway

**Reply by tranb3r** [reply]
You can actually use any remote storage you want.
Firebase Remote Config is just an example. Indeed, the ios bindings are not up-to-date.
But it could be anywhere in the cloud (AWS, Google, Azure...).
It could also be stored on your own backend server, but I do not recommend it, since it could create a chicken-and-egg issue if you have to update both your backend certificate and its certificate pin at the same time.
